### PR TITLE
[CN-62] Add documentation for EC2 to ECS and ECS to EC2 discovery

### PR DESCRIPTION
**Pull request by SeriyBg**

### CODE REVIEW

**Review by hasancelik — APPROVED** [review]

File: docs/modules/deploy/pages/deploying-on-aws.adoc [hunk 1]
```diff
@@ -158,6 +158,9 @@ clientConfig.getNetworkConfig().getAwsConfig()
 --
 ====
 
+Hazelcast Client on EC2 can discover the members running in ECS cluster. For doing so you need specify the needed <<ecs_properties,ECS related properties>> (`cluster`, `family`, `service-name`) and make sure to <<ecs_permissions,required permissions>>.
+If none of the ECS or EC2 related properties are specified, the AWS discovery will try to discover the EC2 memebr first, and then, if no EC2 members found, it will try to discover ECS members.
```
Comment (Serdaro) [inline docs/modules/deploy/pages/deploying-on-aws.adoc:162]: If none of the ECS or EC2 related properties are specified, the AWS discovery tries to discover the EC2 members. If none is found, then it tries to discover ECS members.

File: docs/modules/deploy/pages/deploying-on-aws.adoc [hunk 2]
```diff
@@ -221,19 +224,22 @@ config.getNetworkConfig().getInterfaces().setEnabled(true).addInterface("10.0.*.
...
 
-The following optional properties can be configured:
+The following optional properties can be configured[[ecs_properties]]:
 
 * `access-key`, `secret-key`: access and secret keys of AWS your account; if not set, IAM Task Role is used
 * `region`: region where Hazelcast members are running; default is the current region
 * `cluster`: ECS cluster short name or ARN; default is the current cluster
 * `family`: filter to look only for ECS tasks with the given family name; mutually exclusive with `service-name`
 * `service-name`: filter to look only for ECS tasks from the given service; mutually exclusive with `family`
+* `tag-key`, `tag-value`: filter to look only for ECS Tasks with the given `tag-key`/`tag-value`; multi values supported if comma-separated (e.g. `KeyA,KeyB`); comma-separated values behaves as AND conditions
```
Comment (Serdaro) [inline docs/modules/deploy/pages/deploying-on-aws.adoc:234]: `tag-key`, `tag-value`: filter to look only for ECS Tasks with the given `tag-key`/`tag-value`; comma-separated multiple values are supported, e.g., `KeyA,KeyB`; comma-separated values behave as AND conditions

File: docs/modules/deploy/pages/deploying-on-aws.adoc [hunk 3]
```diff
@@ -300,6 +306,9 @@ clientConfig.getNetworkConfig().getAwsConfig()
 --
 ====
 
+Hazelcast Client on ECS can discover the members running on EC2 instances. For this you need to specify the needed <<ec2_properties,EC2 related properties>> (`iam-role`, `security-group-name`) and make sure to `ec2:DescribeInstances` permission.
```
Comment (Serdaro) [inline docs/modules/deploy/pages/deploying-on-aws.adoc:309]: ... and make sure to set the ...

File: docs/modules/deploy/pages/deploying-on-aws.adoc [hunk 4]
```diff
@@ -300,6 +306,9 @@ clientConfig.getNetworkConfig().getAwsConfig()
...
+If none of the ECS or EC2 related properties are specified, the AWS discovery will try to discover the ECS memebr first, and then, if no ECS members found, it will try to discover EC2 members.
```
Comment (Serdaro) [inline docs/modules/deploy/pages/deploying-on-aws.adoc:310]: If none of the ECS or EC2 related properties are specified, the AWS discovery tries to discover the ECS members. If none is found, then it tries to discover EC2 members.

**Review by Serdaro — COMMENTED** [review]
Minor suggestions.

File: docs/modules/deploy/pages/deploying-on-aws.adoc [hunk 5]
```diff
@@ -158,6 +158,9 @@ clientConfig.getNetworkConfig().getAwsConfig()
 --
 ====
 
+Hazelcast Client on EC2 can discover the members running in ECS cluster. For doing so you need specify the needed <<ecs_properties,ECS related properties>> (`cluster`, `family`, `service-name`) and make sure to <<ecs_permissions,required permissions>>.
```
[the remark this answers is not in the document]
Comment (SeriyBg) [reply docs/modules/deploy/pages/deploying-on-aws.adoc:161]: I have attached the anchor link to the related section of the docs. Please see [line 227](https://github.com/hazelcast/hz-docs/pull/567/files#diff-067307a1b21c06c8eed8add8ca2eb26638ab84199402a8dc9cb5cd8d0c3f9bcaR227) and [line 172](https://github.com/hazelcast/hz-docs/pull/567/files#diff-067307a1b21c06c8eed8add8ca2eb26638ab84199402a8dc9cb5cd8d0c3f9bcaR172)

**Review by Serdaro — APPROVED** [review]

**Comment by Serdaro** [comment]
@SeriyBg , I have a sense that this enhancements also apply to 5.1.5+ and 5.2.x?

**Comment by SeriyBg** [comment]
> @SeriyBg , I have a sense that this enhancements also apply to 5.1.5+ and 5.2.x?

@Serdaro it does apply to 5.2.1 (https://github.com/hazelcast/hazelcast/pull/22590) but not the the 5.1.5+ versions

**Pull request merged**